### PR TITLE
🚀 AdoptAWidget: IgnorePointer

### LINKED ISSUE #69485: Adopt a Widget - IgnorePointer
The [IgnorePointer](https://master-api.flutter.dev/flutter/widgets/IgnorePointer-class.html) widget needs:

- A code / DartPad sample

For details about how to participate, see [flutter.dev/adoptawidget](https://flutter.dev/adoptawidget)

### PR DESCRIPTION
**Pull request by YazeedAlKhalaf**
AdoptAWidget: IgnorePointer

I have added a dartpad and reorganized the explanation.

This pull request is:

- [x] A code sample
- [ ] More references
- [ ] More explanation

closes #69485

### CODE REVIEW

**Comment by JuYeong0413** [comment]
The analyzer is unhappy with the trailing spaces. Could you please check it out? 🙂

**Comment by YazeedAlKhalaf** [comment]
> The analyzer is unhappy with the trailing spaces. Could you please check it out? 🙂

Yup, sure I will! 🚀

**Comment by YazeedAlKhalaf** [comment]
I am trying to find the spaces but the lines it mentions have no trailing spaces, at least in vs code. When I select there is an extra highlight space but I can't remove and it is not space when moving with arrow keys.

File: packages/flutter/lib/src/widgets/basic.dart [hunk 1]
```diff
@@ -6562,6 +6560,64 @@ class RepaintBoundary extends SingleChildRenderObjectWidget {
...
+///             Text(
+///               'Ignoring status: ${ignoring ? 'TRUE' : 'FALSE'}',
+///             ),
+///             ElevatedButton(
+///               onPressed: () {},
+///               child: Text(
+///                 'CLICK ME!',
+///                 style: TextStyle(
+///                   fontSize: 50,
+///                 ),
```
Comment (johnpryan) [inline packages/flutter/lib/src/widgets/basic.dart:6609]: It's a best-practice to use the Material theme for text styles. It's also a little too big :)

```suggestion
///                 style: Theme.of(context).textTheme.body1
```

File: packages/flutter/lib/src/widgets/basic.dart [hunk 2]
```diff
@@ -6562,6 +6560,64 @@ class RepaintBoundary extends SingleChildRenderObjectWidget {
 /// the semantics layer (and thus e.g. accessibility tools). If
 /// [ignoringSemantics] is null, it uses the value of [ignoring].
 ///
+/// {@youtube 560 315 https://www.youtube.com/watch?v=qV9pqHWxYgI}
+///
+/// {@tool dartpad --template=stateful_widget_material}
+/// The following sample has an [IgnorePointer] widget wrapping the `Column`
+/// which contains a button. 
+/// When setting [ignoring] to `true` anything inside the `Column` will
```
Comment (johnpryan) [inline packages/flutter/lib/src/widgets/basic.dart:6568]:
```suggestion
/// When [ignoring] is set to `true` anything inside the `Column` can 
```

File: packages/flutter/lib/src/widgets/basic.dart [hunk 3]
```diff
@@ -6562,6 +6560,64 @@ class RepaintBoundary extends SingleChildRenderObjectWidget {
 /// the semantics layer (and thus e.g. accessibility tools). If
 /// [ignoringSemantics] is null, it uses the value of [ignoring].
 ///
+/// {@youtube 560 315 https://www.youtube.com/watch?v=qV9pqHWxYgI}
+///
+/// {@tool dartpad --template=stateful_widget_material}
+/// The following sample has an [IgnorePointer] widget wrapping the `Column`
+/// which contains a button. 
+/// When setting [ignoring] to `true` anything inside the `Column` will 
+/// not be pressable or tapable. When setting [ignoring] to `false` anything
```
Comment (johnpryan) [inline packages/flutter/lib/src/widgets/basic.dart:6569]:
```suggestion
/// not be tapped. When setting [ignoring] to `false` anything
```

File: packages/flutter/lib/src/widgets/basic.dart [hunk 4]
```diff
@@ -6562,6 +6560,64 @@ class RepaintBoundary extends SingleChildRenderObjectWidget {
...
+/// inside the `Column` will be pressable or tapable.
```
Comment (johnpryan) [inline packages/flutter/lib/src/widgets/basic.dart:6570]:
```suggestion
/// inside the `Column` can be tapped.
```

**Review by johnpryan — CHANGES_REQUESTED** [review]
Looks good overall, just a few suggestions.

**Comment by goderbauer** [comment]
(PR triage) @YazeedAlKhalaf do you have plans to follow up on the feedback?

**Comment by YazeedAlKhalaf** [comment]
> (PR triage) @YazeedAlKhalaf do you have plans to follow up on the feedback?

Yes, but I had so many quizzes last week and this week. Today I am free and will do the changes 🚀.

File: packages/flutter/lib/src/widgets/basic.dart [hunk 5]
```diff
@@ -6562,6 +6560,61 @@ class RepaintBoundary extends SingleChildRenderObjectWidget {
 /// the semantics layer (and thus e.g. accessibility tools). If
 /// [ignoringSemantics] is null, it uses the value of [ignoring].
 ///
+/// {@youtube 560 315 https://www.youtube.com/watch?v=qV9pqHWxYgI}
+///
+/// {@tool dartpad --template=stateful_widget_material}
+/// The following sample has an [IgnorePointer] widget wrapping the `Column`
+/// which contains a button.
+/// When [ignoring] is set to `true` anything inside the `Column` can
```
Comment (dkwingsmt) [inline packages/flutter/lib/src/widgets/basic.dart:6568]: Is this intended to be a new line, or just text wrap? If it's a new line, you should add an empty line. Single empty lines are ignored.

Comment (YazeedAlKhalaf) [reply packages/flutter/lib/src/widgets/basic.dart:6568]: just text wrap

**Review by dkwingsmt — APPROVED** [review]
LGTM

**Comment by dkwingsmt** [comment]
For unknown reasons the tests keep failing (obviously incorrectly). Can you try by merging with the head and pushing again?

**Comment by YazeedAlKhalaf** [comment]
> For unknown reasons the tests keep failing (obviously incorrectly). Can you try by merging with the head and pushing again?

Sure I will

**Comment by YazeedAlKhalaf** [comment]
Can anybody help me with why is the test failing?

**Comment by dkwingsmt** [comment]
I've restarted the tests in case of flakes.

**Review by johnpryan — APPROVED** [review]

**Comment by johnpryan** [comment]
@YazeedAlKhalaf Thanks for participating in AdoptAWidget. We would like to send you a special thanks. If you are interested, please send an email to nyener at google dot com and provide a link to this issue.

**Comment by YazeedAlKhalaf** [comment]
> @YazeedAlKhalaf Thanks for participating in AdoptAWidget. We would like to send you a special thanks. If you are interested, please send an email to nyener at google dot com and provide a link to this issue.

I have sent an email but no reply 😁🚀

**Pull request merged**